src/System.Diagnostics.EventLog/tests/Helpers.cs

```diff
@@ -2,11 +2,89 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.ComponentModel;
+using System.Threading;
+
 namespace System.Diagnostics.Tests
 {
     internal class Helpers
     {
         public static bool IsElevatedAndSupportsEventLogs { get => AdminHelpers.IsProcessElevated() && SupportsEventLogs; }
-        public static bool SupportsEventLogs { get => PlatformDetection.IsNotWindowsNanoServer && !PlatformDetection.IsWindows7; }
+        public static bool SupportsEventLogs { get => PlatformDetection.IsNotWindowsNanoServer; }
+
+        public static void RetryAvailable(Action func)
+        {
+            if (!PlatformDetection.IsWindows7)
+            {
+                func();
+                return;
+            }
+
+            int retries = 3;
+            while (retries > 0)
+            {
+                try
+                {
+                    func();
+                    break;
+                }
+                catch (Win32Exception)
+                {
+                    Thread.Sleep(100);
+                    retries--;
+                }
+            }
+            return;
```

Comment:
I just noticed that if your retries fail, you just return without error. Is this intentional? You may be finding that the tests are stable now not becaues they are successfully calling Delete() after a few retries, but simply because they are catching the exception. I would add Assert.True(retries > 0);

Reply:
You may then find you need more retries to be stable.